src/js/Setup.js

```diff
@@ -5,8 +5,46 @@ var cslib;
 
 var cscompiled = {};
 
+// Simulation settings
 var csanimating = false;
 var csstopped = true;
+var simtime = 0; // accumulated simulation time since start
+var simspeed = 0.5; // global speed setting, will get scaled for applications
+var simcap = 1000 / 20; // max. ms between frames for fps-independent sim
+var simtick = 0; // Date.now of the most recent simulation tick
+var simaccuracy = 10; // number of sub-steps per frame
+
+var simunit = 5 / 360; // reported simulationtime() per internal simtime unit
+/* Cinderella has a factor 5 for its internal animation clock,
+ * and the division by 360 is in the simulationtime function implementation.
+ */
+
+// internal simtime units per millisecond at simspeed 1
+var simfactor = 0.32 / simunit / 1000 * 2;
```

Comment:
This appears to be more of a problem with legacy widgets. In particular, when I changed the default speed from 1 to 0.5 I forgot to change the scale factor for backwards compatible time control (via the deltat parameter of the environment). My latest commit should take care of that.

But this only strengths the importance of the question I asked above: do we actually want this kind of backwards-compatibility (and maintain it throughout the 0.8 cycle at the least), or do we want to remove that now and force widgets using 0.8 to set speed in a modern way?